lib/src/model/game/game_repository.dart

```diff
@@ -42,6 +42,17 @@ class GameRepository {
     );
   }
 
+  Future<FullGamePaginator> getFullGames(
+    UserId userId,
+    int page,
+  ) {
+    return client.readJson(
+      Uri.parse('$kLichessHost/@/$userId/all?page=$page'),
```

Comment:
This endpoint is not opened to OAuth. I'm not sure we should use it since it's an old one not part of the API, and it would be rate limited.

I think https://lichess.org/api#tag/Games/operation/apiGamesUser has everything we need.

cc @ornicar , what do you think about that?

Reply:
I did consider using this API, but it doesn't fit with `LightGame` which means adding custom implementation many things like the game tiles. Where as the old API just works. It would be nice if the new API can have the same response format as recent games.

Reply:
I don't understand @ZTL-UwU 
This API does perfectly fit with `LightArchivedGame`, it is already used in the code, here: https://github.com/lichess-org/mobile/blob/d8b63a1d0afde6b54a57f31cba4279065598c9c3/lib/src/model/game/game_repository.dart#L41

Reply:
@veloce Sorry, I've got myself confused, it was written quite some time ago. What really happened was that the API doesn't have pagination, there is only a `max` query so I had to get all the games at once.

↓ This was the comment in this PR previously.
> Currently we are downloading all game histories at once, which takes a very long time to load on startup if the user has a large number of games. We will need a pagination API to get a few games everytime.

The API documentation says _We recommend streaming the response_. But I don't know how it will work with an infinite scroll so I used the old API instead in 78ff233c0c5c6471b91592512b90a6dcc04a72f1.

Reply:
Actually you can paginate using the new endpoint, using `max` and `until`.

You already have an example in the app with the puzzle activity API which works the same way:
- https://lichess.org/api#tag/Puzzles/operation/apiPuzzleActivity
- https://github.com/lichess-org/mobile/blob/e56f52fd8751bb7ca2d8f404cba5b07ce5675c64/lib/src/model/puzzle/puzzle_activity.dart#L63